File: Python/python/Prototype/Rectangle.py
import copy
from .ShapePrototype import ShapePrototype

# Implementación concreta del prototipo: Rectángulo
class Rectangle(ShapePrototype):
    def __init__(self, width: int, height: int):
        self._width = width
        self._height = height
        self._color = None

    def clone(self):
        return copy.deepcopy(self)

    def draw(self):
        print(f"Dibujando rectángulo - Ancho: {self._width}, Alto: {self._height}, Color: {self._color}")

    def set_color(self, color: str):
        self._color = color

    @property
    def color(self):
        return self._color

File: src/prototype/Rectangle.java
package prototype;

public class Rectangle implements ShapePrototype{
    private String color;
    private int width;
    private int height;

    public Rectangle(int width, int height) {
        this.width = width;
        this.height = height;
    }

    @Override
    public ShapePrototype clone() {
        Rectangle clone = new Rectangle(this.width, this.height);
        clone.setColor(this.color);
        return clone;
    }

    @Override
    public void draw() {
        System.out.println("Dibujando rectángulo - Ancho: " + width
                + ", Alto: " + height + ", Color: " + color);
    }

    @Override
    public void setColor(String color) {
        this.color = color;
    }

    @Override
    public String getColor() {
        return color;
    }
}
